src/App.js

```diff
@@ -30,128 +30,69 @@ export default class App extends Component {
+        currentLang,
+        hasLanguageChanged: this.state.hasLanguageChanged + 1,
+      });
+    },
+    setImageType: (imageType) => {
+      this.setState({
+        imageType,
+      });
+    },
+    showError: (error) => {
+      console.error(error);
       this.setState(({ errors }) => ({
-        errors: errors.slice(1),
+        errors: errors.concat(error),
       }));
-    }, 2500);
-  };
-
-  showInfo = ({ id, message }) => {
-    this.setState(({ infos }) => ({
-      infos: infos.concat({ id, message }),
-    }));
-    setTimeout(() => {
+      setTimeout(() => {
+        this.setState(({ errors }) => ({
+          errors: errors.slice(1),
+        }));
+      }, 2500);
+    },
+    showInfo: ({ id, message }) => {
       this.setState(({ infos }) => ({
-        infos: infos.slice(1),
+        infos: infos.concat({ id, message }),
       }));
-    }, 2500);
+      setTimeout(() => {
+        this.setState(({ infos }) => ({
+          infos: infos.slice(1),
+        }));
+      }, 2500);
+    },
+    setShowBirthName: (showBirthName) => {
+      this.setState({ showBirthName });
+    },
+    setShowNavIcons: (showNavIcons) => {
+      this.setState({ showNavIcons });
+    },
+    setShowFace: (showFace) => {
+      this.setState({ showFace });
+    },
   };
 
-  setShowGenderColor = (showGenderColor) => {
-    this.setState({ showGenderColor });
-  };
-  setShowBirthName = (showBirthName) => {
-    this.setState({ showBirthName });
-  };
-  setShowNavIcons = (showNavIcons) => {
-    this.setState({ showNavIcons });
-  };
-  setShowFace = (showFace) => {
-    this.setState({ showFace });
+  componentDidCatch = (error) => {
+    this.state.showError(error);
   };
+
   render() {
-    const {
-      showError,
-      showInfo,
-      setCurrentLang,
-      setCurrentProp,
-      setCurrentEntity,
-      setShowGenderColor,
-      setShowBirthName,
-      setShowNavIcons,
-      setShowFace,
-      setLoadingEntity,
-      setImageType,
-    } = this;
-    const {
-      currentLang,
-      errors,
-      infos,
-      currentEntity,
-      currentProp,
-      hasLanguageChanged,
-      showGenderColor,
-      showBirthName,
-      showNavIcons,
-      showFace,
-      imageType,
-      loadingEntity,
-    } = this.state;
+    const { errors, infos } = this.state;
     return (
-      <AppContext.Provider
-        value={{
-          showError,
-          showInfo,
-          setCurrentLang,
-          setCurrentProp,
-          setCurrentEntity,
-          setShowGenderColor,
-          setShowBirthName,
-          setShowFace,
-          setShowNavIcons,
-          setImageType,
-          setLoadingEntity,
-          currentLang,
-          currentProp,
-          currentEntity,
-          hasLanguageChanged,
-          showGenderColor,
-          showBirthName,
-          showNavIcons,
-          showFace,
-          imageType,
-          loadingEntity,
-        }}
-      >
+      <AppContext.Provider value={this.state}>
```

Comment:
this way the object doesn't get created every time a variable in the state changes